LCM/dsc/engine/ConfigurationManager/LocalConfigManagerHelper.c

```diff
@@ -674,7 +674,17 @@ MI_Result CallTestConfiguration(
 
     SetMessageInContext(ID_OUTPUT_OPERATION_START,ID_OUTPUT_ITEM_TEST,&lcmContext);
     LCM_BuildMessage(&lcmContext, ID_OUTPUT_EMPTYSTRING, EMPTY_STRING, MI_WRITEMESSAGE_CHANNEL_VERBOSE);
-    result = ApplyCurrentConfig(&lcmContext, moduleManager, LCM_EXECUTE_TESTONLY, &resultStatus, cimErrorDetails);
+    
+    if (File_ExistT(GetCurrentConfigFileName()) != -1)
+    {
+        result = ApplyCurrentConfig(&lcmContext, moduleManager, LCM_EXECUTE_TESTONLY, &resultStatus, cimErrorDetails);
+    }
+
+    if (File_ExistT(GetPendingConfigFileName()) != -1)
+    {
+        DSC_WriteWarning(&lcmContext, ID_LCMHELPER_TEST_OPERATION_AGAINST_PENDING);
+        result = ApplyPendingConfig(&lcmContext, moduleManager, LCM_EXECUTE_TESTONLY, &resultStatus, cimErrorDetails);
+    }
```

Comment:
Is it possible with new changes that current and pending both exist on the machine. scenario 
1- where you apply one configuration it sets fine and file is current.mof
2- you apply new configuration now it fails is it stored as pending now ?
3- if (2) is true then here we are not doing the right thing. we are applying both current and pending. Reason it is okay for windows as it is not possible to have both pending & current present at same time on windows. 
4- if (2) is not correct then new configuration will be set as failed configuration. which means get and set will continue working with current.mof which is not correct behavior either as once (2) is done  by user to set configuration the sate of machine is represented by new mof document in (2) not what was on machine at point (1)

Reply:
When you apply a new configuration it is saved as Pending.mof. If the application of that configuration fails, Pending.mof is copied to Failed.mof and Pending.mof is removed. Current.mof is not affected.

So the change that is needed is that Current.mof should be moved to Previous.mof before Pending.mof is applied (and can possibly fail). This was previously only done if Pending.mof succeeded.

I have made this change by placing the move from Current.mof to Previous.mof in all the same places as in the Windows code.
I am re-testing the Linux code now with the change.